### ISSUE TITLE
Doesn't work with phone in airplane mode

### ISSUE THREAD

**Issue by Notmarrco** [mclear/Android_NFC_Ring_Unlock#52]
Airplane mode disables nfc (at least on cyanogen), so the unlock app should check the nfc status and enable it if needed imo.

**Comment by khimera80** [comment]
I second this. Luckily, I had a passcode entered for emergencies, but this one almost bit me in the a**.

**Comment by r2DoesInc** [comment]
Not possible to modify NFC settings directly. See http://stackoverflow.com/a/5961338/1117029

**Comment by khimera80** [comment]
Maybe it would be possible to add a setting that says "don't lock screen if airplane mode enabled"?

**Comment by r2DoesInc** [comment]
@JohnMcLear will have to make the decision on how this should be handled. A setting for disabling while in airplane or while nfc is off is not a bad idea.

**Comment by khimera80** [comment]
Thanks.

Extended idea: Disable if NFC disabled AND no pin code is set. If PIN code is set, then I probably would use that to unlock ;)

**Comment by JohnMcLear** [comment]
I'd say disabling NFC Unlock while in flight mode would be sensible if possible.  

The risk here is that It does mean that someone could pick up your phone, change it into flight mode and not have to worry about any security.  

I can't see any other way to win though..

I think we need the option but I'm not sure if we should have it on by default or not, we certainly don't want people being locked out of their device which is the current status quo...

**Comment by r2DoesInc** [comment]
requiring a pin will circumvent this.

**Comment by JohnMcLear** [comment]
Agreed, so maybe we just ensure any NFC Ring users have 3 forms of unlock..

NFC / Pin / Google login

**Comment by r2DoesInc** [comment]
Gotta figure out how to make that super clear and guide the user through
the setup of all three.

also, uber app lock now has a pattern unlock, im bringing it into nfcsecure
soon, something to consider.

**Comment by jezmck** [comment]
+1 for pattern unlock

**Comment by Notmarrco** [comment]
according to the stackoverflow link it should be possible to activate nfc when root. 
Since I'm root I'd like that :) 
and +1 for pattern unlock

**Comment by r2DoesInc** [comment]
The app will NOT be implementing ANY root solutions. This is absolutely NOT
a viable solution.

**Comment by jezmck** [comment]
Again, that's a very stubborn (rude) response.

Perhaps the app can be forked to allow a root-enabled version?

On 30 April 2014 13:03, Ken notifications@github.com wrote:

> The app will NOT be implementing ANY root solutions. This is absolutely
> NOT
> a viable solution.
> On Apr 30, 2014 7:55 AM, "Notmarrco" notifications@github.com wrote:
> 
> > according to the stackoverflow link it should be possible to activate
> > nfc
> > when root.
> > Since I'm root I'd like that :)
> > and +1 for pattern unlock
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/mclear/Android_NFC_Ring_Unlock/issues/52#issuecomment-41787864>
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/mclear/Android_NFC_Ring_Unlock/issues/52#issuecomment-41788493
> .

## 

Jez McKean
_jezmckean.com http://jezmckean.com_

**Comment by r2DoesInc** [comment]
Sorry, but again, root code will not be introduced into the app. Its not
about being stubborn, the app was originally root only and I have done
everything in my power to create an u lock method for non rooted users. If
you have root and don't mind using that, there are WAY better NFC unlock
options out there. This apps ONLY focus is to bring this behavior to the
masses as a completely user space app using only well documented, public
APIs.

**Comment by r2DoesInc** [comment]
Check out this app if you are looking for a root solution. If this type of
behavior was possible with this app, I would have written it that way to
begin with, sadly, as a 100% user space app, we are limited to what the OS
allows us. With a root app, there are no such limitations.

https://play.google.com/store/apps/details?id=com.steelrat.nfcunlocker

**Comment by jezmck** [comment]
Will do, thank you.

On 30 April 2014 13:38, Ken notifications@github.com wrote:

> Check out this app if you are looking for a root solution. If this type of
> behavior was possible with this app, I would have written it that way to
> begin with, sadly, as a 100% user space app, we are limited to what the OS
> allows us. With a root app, there are no such limitations.
> 
> https://play.google.com/store/apps/details?id=com.steelrat.nfcunlocker
> On Apr 30, 2014 8:29 AM, "Ken Kyger" r2doesinc@gmail.com wrote:
> 
> > Sorry, but again, root code will not be introduced into the app. Its not
> > about being stubborn, the app was originally root only and I have done
> > everything in my power to create an u lock method for non rooted users.
> > If
> > you have root and don't mind using that, there are WAY better NFC unlock
> > options out there. This apps ONLY focus is to bring this behavior to the
> > masses as a completely user space app using only well documented, public
> > APIs.
> > On Apr 30, 2014 8:25 AM, "Jez McKean" notifications@github.com wrote:
> > 
> > > Again, that's a very stubborn (rude) response.
> > > 
> > > Perhaps the app can be forked to allow a root-enabled version?
> > > 
> > > On 30 April 2014 13:03, Ken notifications@github.com wrote:
> > > 
> > > > The app will NOT be implementing ANY root solutions. This is
> > > > absolutely
> > > > NOT
> > > > a viable solution.
> > > > On Apr 30, 2014 7:55 AM, "Notmarrco" notifications@github.com
> > > > wrote:
> > > > 
> > > > > according to the stackoverflow link it should be possible to
> > > > > activate
> > > > > nfc
> > > > > when root.
> > > > > Since I'm root I'd like that :)
> > > > > and +1 for pattern unlock
> > > > > 
> > > > > —
> > > > > Reply to this email directly or view it on GitHub<
> > > 
> > > https://github.com/mclear/Android_NFC_Ring_Unlock/issues/52#issuecomment-41787864>
> > > 
> > > > > .
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub<
> > > 
> > > https://github.com/mclear/Android_NFC_Ring_Unlock/issues/52#issuecomment-41788493>
> > > 
> > > > .
> > > 
> > > ## 
> > > 
> > > Jez McKean
> > > _jezmckean.com http://jezmckean.com_
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub<
> > > https://github.com/mclear/Android_NFC_Ring_Unlock/issues/52#issuecomment-41790122>
> > > 
> > > .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/mclear/Android_NFC_Ring_Unlock/issues/52#issuecomment-41791251
> .

## 

Jez McKean
_jezmckean.com http://jezmckean.com_

**Comment by JohnMcLear** [comment]
Yea I agree with @r2DoesInc we don't want the app to require rooting your phone, that defeats the purpose.

**Comment by KevinCBoyle** [comment]
This is really nasty.  There is nothing in the setup that warns you to set a PIN (or nothing I saw) and NO warning that without a PIN that Google password won't work to unlock once you shut off the radio.  I agree no rooting, but this is a sangerous trap (that I confess I've fallen into).

:(

**Comment by EduardFriesen** [comment]
+1 for pattern

**Comment by Lokki-P** [comment]
+1 for mandatory alternate unlock method. Guides in the forum are updated to include PIN in initial setup.
